src/components/validation/AddValidation.js

```diff
@@ -0,0 +1,440 @@
+    const schemaIds = datasets.find(({ id }) => id === sourceId)?.schemaIds;
+    if (schemaIds?.length && !schemaIds?.includes(schemaId)) {
+      [schemaId] = schemaIds;
+      this.setState({
+        schemaId,
+      });
+    }
+  };
+
+  handleExpand = (idx) => (event, isExpanded) => {
+    this.setState({ expandedIdx: isExpanded ? idx : false });
+  };
+
+  handleParametersOnChange = (parameters, algoIdx) => {
+    const { selectedValidationAlgorithms } = this.state;
+    const updatedAlgo = {
+      ...selectedValidationAlgorithms[algoIdx],
+      parameters,
+    };
+
+    const updatedAlgorithms = selectedValidationAlgorithms.slice();
+    updatedAlgorithms.splice(algoIdx, 1, updatedAlgo);
+    this.setState({ selectedValidationAlgorithms: updatedAlgorithms });
+  };
+
+  handleSchemaOnChange = (schemaId) => {
+    this.setState({ schemaId });
+  };
+
+  handleRemove = (idx) => {
+    const { selectedValidationAlgorithms } = this.state;
+
+    const newAlgos = selectedValidationAlgorithms.slice();
+    newAlgos.splice(idx, 1);
+
+    this.setState({
+      selectedValidationAlgorithms: newAlgos,
+      expandedIdx: false,
+    });
+  };
+
+  handleExecute = () => {
+    const {
+      history: { goBack },
+      dispatchCreateValidation,
+    } = this.props;
+    const { sourceId, selectedValidationAlgorithms, schemaId } = this.state;
+    dispatchCreateValidation({
+      sourceId,
+      algorithms: selectedValidationAlgorithms,
+      schemaId,
+    });
+    goBack();
+  };
+
+  handleAddAlgorithm = () => {
+    const { validationAlgorithms } = this.props;
+    const { algorithmId, selectedValidationAlgorithms } = this.state;
+    const selectedAlgo = validationAlgorithms.find(
+      ({ id }) => id === algorithmId,
+    );
+    if (selectedAlgo) {
+      const newSelectedAlgos = selectedValidationAlgorithms.slice();
+      newSelectedAlgos.push(selectedAlgo);
+      this.setState({
+        selectedValidationAlgorithms: newSelectedAlgos,
+        expandedIdx: newSelectedAlgos.length - 1,
+        algorithmId: '',
+      });
+    }
+  };
+
+  handleAlgorithmSelectOnChange = (e) => {
+    const algorithmId = e.target.value;
+    this.setState({
+      algorithmId,
+    });
+  };
+
+  renderDatasetsAndResultsSelect = () => {
+    const { sourceId } = this.state;
+    const { classes, t, datasets, results, schemas } = this.props;
+
+    const datasetMenuItems = datasets
+      .sortBy(({ name }) => name)
+      .map(({ id, name, schemaIds }) => (
+        <MenuItem
+          id={buildAddValidationDatasetOptionId(id)}
+          value={id}
+          key={id}
+          className={classes.menuItem}
+        >
+          <Grid container alignItems="center" spacing={1}>
+            <Grid item>{name}</Grid>
+            {schemaIds?.map((schemaId) => (
+              <Grid item key={schemaId}>
+                <SchemaTag schema={schemas.get(schemaId)} />
```

Comment:
Would it make sense for `SchemaTag` to process the `schemaId` itself? It would fetch `schemas` from the database and have `schemaId` as prop.

Reply:
I think it would make sense. The only issue is the `...` schema tag that shows up when there are more than 3 schemas. It is defined on the fly and is not in `schemas`. 

possible workarounds:
- have a different component for the `...` schema tag
- deal with it through the props somehow. Maybe check if id is some predefined constant

What do you think?

Reply:
Isn't it possible to check depending on the number of tags? We can also create a whole component `SchemaTags` which receive the schemaIds and deal with the `...`? We can discuss live about it.

Reply:
Isn't it possible to check depending on the number of tags? We can also create a whole component SchemaTags which receive the schemaIds and deal with the ...? We can discuss live about it.